src/vs/platform/terminal/node/terminalProcess.ts

```diff
@@ -111,15 +120,24 @@ export class TerminalProcess extends Disposable implements ITerminalChildProcess
 			conptyInheritCursor: useConpty && !!_shellLaunchConfig.initialText
 		};
 		// Delay resizes to avoid conpty not respecting very early resize calls
-		if (platform.isWindows && useConpty && cols === 0 && rows === 0 && this._shellLaunchConfig.executable?.endsWith('Git\\bin\\bash.exe')) {
-			this._delayedResizer = new DelayedResizer();
-			this._register(this._delayedResizer.onTrigger(dimensions => {
-				this._delayedResizer?.dispose();
-				this._delayedResizer = undefined;
-				if (dimensions.cols && dimensions.rows) {
-					this.resize(dimensions.cols, dimensions.rows);
+		if (platform.isWindows) {
+			if (useConpty && cols === 0 && rows === 0 && this._shellLaunchConfig.executable?.endsWith('Git\\bin\\bash.exe')) {
+				this._delayedResizer = new DelayedResizer();
+				this._register(this._delayedResizer.onTrigger(dimensions => {
+					this._delayedResizer?.dispose();
+					this._delayedResizer = undefined;
+					if (dimensions.cols && dimensions.rows) {
+						this.resize(dimensions.cols, dimensions.rows);
+					}
+				}));
+			}
+			this.onProcessReady(() => {
+				if (this._ptyProcess?.pid) {
```

Comment:
The `this.onProcessReady` event actually  contains pid, we can avoid the `?` like:

```ts
this.onProcessReady(e => {
  this._windowsShellHelper = this._register(new WindowsShellHelper(e.pid));
```